narwhal/network/src/connectivity.rs

```diff
@@ -97,13 +93,18 @@ impl ConnectionMonitor {
         peer_id: PeerId,
         connection_status: ConnectionStatus,
     ) {
-        self.connection_metrics.network_peers.inc();
+        let int_status = match connection_status {
```

Comment:
Just to make sure, it is impossible for a given peer to call this function on a disconnected status first?

Reply:
Thanks for pointing this out @mwtian  - anemo will always emit a connected event first for a peer before emitting a disconnected one, so that will allow us to keep the metric consistent. However, I see your point and we can play it a bit more safe here - I'll update the metric based on current instance of active peers.